src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -392,20 +388,48 @@ public bool MoveNext()
             {
                 if (_version != _q._version) throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
 
-                if (_index == -2)
-                    return false;
-
-                _index++;
+                // Instead of return true/false, we assign the result to a variable and
+                // return it at the end. This helps decrease code size, as currently the
+                // jit cannot do this for us and generates code for 3 returns.
+                bool result = false;
```

Comment:
Let's keep this PR to exactly what it was meant to be about: removing the modulo.  If it's then justifiable to do further surgery, that can be done separately.